functions/src/handlers/extractions.ts

```diff
@@ -0,0 +1,554 @@
+    nameField: 'Area_name',
+  },
+  stream: {
+    url: 'https://services1.arcgis.com/99lidPhWCzftIe9K/ArcGIS/rest/services/UtahStreamsNHD/FeatureServer/0',
+    attributes: ['FCode_Text'] as const,
+    nameField: 'FCode_Text',
+  },
+} as const;
+export type LayerName = keyof typeof FEATURE_SERVICE_CONFIG;
+export const SPATIAL_REFERENCES = {
+  WEB_MERCATOR: 3857,
+  UTM_ZONE_12N: 26912,
+} as const;
+export interface LayerCriteria {
+  attributes: string[];
+}
+export type ExtractionCriteria = Partial<Record<LayerName, LayerCriteria>>;
+export interface IntersectionResult {
+  [key: string]: string | number;
+  size: number;
+  displaySize: string;
+}
+export type IntersectionResponse = Partial<Record<LayerName, IntersectionResult[]>>;
+export type QueryResponse = Pick<__esri.FeatureSetProperties, 'features' | 'geometryType' | 'spatialReference'> & {
+  features: FeatureJSON[];
+};
+export interface AreasAndLengthsResponse {
+  areas?: number[];
+  lengths?: number[];
+}
+
+const numberFormatter = new Intl.NumberFormat('en-US', {
+  minimumFractionDigits: 2,
+  maximumFractionDigits: 2,
+});
+
+/**
+ * Queries an ArcGIS feature service for features intersecting a geometry
+ * @param serviceUrl - URL of the feature service layer
+ * @param geometry - Input geometry to query against
+ * @param outFields - Array of field names to return
+ * @returns Query response with features
+ */
+export async function queryFeatureService(
+  serviceUrl: string,
+  geometry: GeometryJSON,
+  outFields: string[],
+): Promise<QueryResponse> {
+  const params = new URLSearchParams({
+    f: 'json',
+    geometry: JSON.stringify(geometry),
+    geometryType: getGeometryType(geometry),
+    spatialRel: 'esriSpatialRelIntersects',
+    outFields: outFields.join(','),
+    returnGeometry: 'true',
+    outSR: String(SPATIAL_REFERENCES.UTM_ZONE_12N),
+  });
+
+  const url = `${serviceUrl}/query?${params.toString()}`;
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    throw new Error(`Feature service query failed: ${response.status} ${response.statusText}`);
+  }
+
+  const data = (await response.json()) as __esri.FeatureSetProperties | { error: { message?: string } };
+
+  if ('error' in data) {
+    throw new Error(`Feature service error: ${data.error.message || JSON.stringify(data.error)}`);
+  }
+
+  return data as QueryResponse;
+}
+
+/**
+ * Determines the Esri geometry type from a geometry object
+ */
+function getGeometryType(geometry: GeometryJSON): string {
+  switch (geometry.type.toLowerCase()) {
+    case 'polygon':
+      return 'esriGeometryPolygon';
+    case 'polyline':
+      return 'esriGeometryPolyline';
+    case 'point':
+      return 'esriGeometryPoint';
+    default:
+      throw new Error(`Unsupported geometry type: ${geometry.type}`);
+  }
+}
+
+/**
+ * Converts Esri JSON geometry to ArcGIS SDK geometry
+ */
+function esriJsonToGeometry(esriJson: GeometryJSON, spatialReference: SpatialReference): Polygon | Polyline | Point {
```

Comment:
Could this function be replaced by [fromJSON](https://developers.arcgis.com/javascript/latest/api-reference/esri-geometry-support-jsonUtils.html#fromJSON)?